src/sentry/migrations/0191_make_externalactor_integration_id_not_null.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 1.11.29 on 2021-05-03 22:09
+
+import django.db.models.deletion
+from django.db import migrations
+
+import sentry.db.models.fields.foreignkey
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0190_drop_external_user_table"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            state_operations=[
+                migrations.AlterField(
+                    model_name="externalactor",
+                    name="integration",
+                    field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        default=1,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="sentry.Integration",
+                    ),
+                    preserve_default=False,
+                ),
+            ],
+            database_operations=[],
```

Comment:
You'll need to include a `RunSQL` here with
```
ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" SET DEFAULT 1;
ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" SET NOT NULL;
ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" DROP DEFAULT;
```

(Just to be sure I'm right, could you test this locally and make sure an `ExternalActor` with a null `integration_id` is correctly set to a the default of 1?)

Reply:
got it, was just poking around for how to include that. let me run it locally

Reply:
Update - this didn't work, trying to figure it out. It's upset about the null values

`psycopg2.IntegrityError: IntegrityError('column "integration_id" contains null values\n',)`

Reply:
The original DDL had an update statement in it as well, so you could add that back in. I confused myself here - if you add a column with a default, Postgres will backfill it for you. But if you add a default to the column afterwards it only applies to new rows.
```
ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" SET DEFAULT 1;
UPDATE sentry_externalactor SET integration_id = 1 where integration_id is  null;
ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" SET NOT NULL;
ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" DROP DEFAULT;
```

Reply:
lol I had tried that too 😭 but I get:

`django.db.utils.OperationalError: OperationalError('cannot ALTER TABLE "sentry_externalactor" because it has pending trigger events\n',)`

I'm reading through https://code.djangoproject.com/ticket/23609 and this seems to work:

```
                    ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" SET DATA TYPE integer USING coalesce("integration_id", 1);
                    ALTER TABLE "sentry_externalactor" ALTER COLUMN "integration_id" SET NOT NULL;
```

Reply:
Hmm, based on https://stackoverflow.com/questions/12838111/django-db-migrations-cannot-alter-table-because-it-has-pending-trigger-events it might be just because we're in a transaction. Try setting `atomic=False` and see if the update works

Reply:
with `atomic=False` I get the same error as above. (also learned why you have to set `reverse_sql` 🙃 )

Reply:
I guess it's because we're doing things in the same `RunSQL`, maybe that counts as the same transaction even without atomic. You could try splitting the statements up. I don't really want to use the coalesce method since it's a bit hacky, haha.

Reply:
wooo got it. I'm not positive my `reverse_sql` is perfect but it did work locally